### DISCUSSION TITLE
How can I recursively add packagedeps?

### DISCUSSION THREAD

**Discussion by xq114** [xmake-io/xmake#4430] [Q&A]
```
import("package.tools.meson").install(package, configs, {packagedeps = {"glib"}})
```
会把glib加入flag中；但glib依赖的其他库（libintl, libiconv等）没有加，导致仍然会报找不到符号，需要一个一个把依赖库分不同平台加上去。有没有办法指定添加glib同时把它依赖的库也加到flag中？

**Comment by waruqi** [comment]
package:dep("glib"):librarydeps() 可以取到所有。。但也只能外面加，不过不需要分平台。

```lua
local packagedeps = {"glib"}
for _, name in ipairs(package:dep("glib"):librarydeps()) do
    table.insert(packagedeps, name)
end
```

**Reply by xq114** [reply]
但这里有个问题，这里面的库如果没有加到add_deps里面，那packagedeps就不会生效，meson.lua里面实现的时候是直接取的package:dep("name")。可能还是得在on_load里面传递一下？

**Reply by waruqi** [reply]
add_deps 里肯定还是要加的，不管是 meson/cmake 的tools 里面，都是走 package:dep("name") 取的。。

packagedeps 的作用是  add_deps 加了后 meson/cmake 还是无法自动找到，只能强行通过 packagedeps 打入 flags ，所以前提就是 add_deps 肯定已经在了
